webui/react/src/pages/Clusters.tsx

```diff
@@ -56,24 +59,33 @@ const Clusters: React.FC = () => {
   );
 
   const tabItems: TabsProps['items'] = useMemo(() => {
-    return [
+    const clustersOverview: Readonly<TabsProps['items']> = [
       { children: <ClustersOverview />, key: TabType.Overview, label: 'Overview' },
+    ];
+    const historicalUsage: Readonly<TabsProps['items']> = [
       {
-        children: (
-          <div className={css.tab}>
-            <ClusterHistoricalUsage />
-          </div>
-        ),
+        children: <ClusterHistoricalUsage />,
         key: TabType.HistoricalUsage,
         label: 'Historical Usage',
       },
-      {
-        children: <ClusterLogs />,
-        key: TabType.Logs,
-        label: 'Master Logs',
-      },
     ];
-  }, []);
+    const masterLogs: Readonly<TabsProps['items']> = [
+      { children: <ClusterLogs />, key: TabType.Logs, label: 'Master Logs' },
+    ];
+    const tabs: TabsProps['items'] = [];
+
+    if (rbacEnabled) {
+      tabs.push(...clustersOverview);
+      if (canAdministrateUsers) {
+        tabs.push(...historicalUsage, ...masterLogs);
+      }
+    } else {
+      // if RBAC is not enabled, show all tabs
+      tabs.push(...clustersOverview, ...historicalUsage, ...masterLogs);
+    }
```

Comment:
nit: If you remove the '[]' when defining individual tabs, you don't have to use the spread operator in push

Reply:
i couldn't find `Tab` type from antd. `TabsProps['items']` is `Tab[]` alias. 
i think the current way is safer, but i agree that spread operator isnt cool. 
i'll leave it as it is now.

Reply:
i actually found a way to remove spread operator.